File: Server/src/Server.Application/ViewModels/ReviewViewModel.cs
namespace Server.Application.ViewModels;

public sealed class ReviewViewModel
{
    public Guid Id { get; set; }
    public string? Title { get; set; }
    public string? Description { get; set; }
    public double Rating { get; set; }
    public ReviewUserViewModel User { get; set; } = null!;
    public DateTime Created { get; set; }
    public DateTime? Updated { get; set; }
}

File: client/src/models/reviews/ReviewViewModel.ts
import { z } from 'zod';
import ReviewUserViewModel from './ReviewUserViewModel';

const ReviewViewModel = z.object({
  id: z.string().uuid(),
  title: z.string().max(200).nullable(),
  description: z.string().nullable(),
  rating: z.number(),
  isCriticRating: z.boolean(),
  user: ReviewUserViewModel,
  created: z.string(),
});

export default ReviewViewModel;
export type ReviewViewModelType = z.infer<typeof ReviewViewModel>;
